### PR TITLE
ddl: enhance validation of column names when creating table

### PR DESCRIPTION
**Pull request by ciscoxll**
Check the table name that are included in the field is it consistent with the `create table name`
respond issues #6330

```
mysql:
mysql> drop table if exists t; create table t(c1.c2 blob default null); insert into t values(); select * from t;
Query OK, 0 rows affected, 1 warning (0.00 sec)

ERROR 1103 (42000): Incorrect table name 'c1'
ERROR 1146 (42S02): Table 'test.t' doesn't exist
ERROR 1146 (42S02): Table 'test.t' doesn't exist
```

```
tidb:
tidb> drop table if exists t; create table t(c1.c2 blob default null); insert into t values(); select * from t;
Query OK, 0 rows affected (0.00 sec)

ERROR 1103 (42000): Incorrect table name 'c1'
ERROR 1146 (42S02): Table 'test.t' doesn't exist
ERROR 1146 (42S02): Table 'test.t' doesn't exist
```

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -536,6 +536,15 @@ func checkTooManyColumns(colDefs []*ast.ColumnDef) error {
 	return nil
 }
 
+func checkContainDotColumn(tableName string, colDefs []*ast.ColumnDef) error {
```
Comment (shenli) [inline ddl/ddl_api.go]: Add comments for the function.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -536,6 +536,16 @@ func checkTooManyColumns(colDefs []*ast.ColumnDef) error {
 	return nil
 }
 
+// checkContainDotColumns check field contains the table name.
+func checkContainDotColumns(tableName string, colDefs []*ast.ColumnDef) error {
```
Comment (zimulala) [inline ddl/ddl_api.go]: Could we check it in the function of `Preprocess`?

File: plan/preprocess.go [hunk 3]
```diff
@@ -462,6 +467,16 @@ func isIncorrectName(name string) bool {
 	return false
 }
 
+// checksContainDotColumns check field contains the table name.
```
Comment (XuHuaiyu) [inline plan/preprocess.go]: This comment is not that explicit, better put an example here?

File: executor/executor_test.go [hunk 4]
```diff
@@ -2515,6 +2515,15 @@ func (s *testSuite) TestIssue5341(c *C) {
 	tk.MustQuery("select * from test.t where a < 1 order by a limit 0;").Check(testkit.Rows())
 }
 
+func (s *testSuite) TestContainDotColumn(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists test.t1")
+	tk.MustExec("create table test.t1(t1.a char)")
+	tk.MustExec("drop table if exists t2")
+	tk.MustExec("create table t2(a char, t2.b int)")
```
Comment (XuHuaiyu) [inline executor/executor_test.go]: add test for 
create table t(s.a char);

File: plan/preprocess.go [hunk 5]
```diff
@@ -224,6 +224,11 @@ func (p *preprocessor) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
 		return
 	}
 
+	if tableName, ok := checksContainDotColumns(tName, stmt.Cols); ok {
```
Comment (winkyao) [inline plan/preprocess.go]: add some comments

**Comment by winkyao** [comment]
Please address comments

File: executor/executor_test.go (hunk 4 above)
Comment (ciscoxll) [reply executor/executor_test.go]: Already add.

File: plan/preprocess.go [hunk 6]
```diff
@@ -52,6 +52,7 @@ func (p *preprocessor) Enter(in ast.Node) (out ast.Node, skipChildren bool) {
 	case *ast.CreateTableStmt:
 		p.inCreateOrDropTable = true
 		p.checkCreateTableGrammar(node)
+		p.checkContainDotColumn(node)
```
Comment (winkyao) [inline plan/preprocess.go]: call it in p.checkCreateTableGrammar is better.

**Comment by ciscoxll** [comment]
@winkyao @zimulala @XuHuaiyu PTAL

File: plan/preprocess.go [hunk 7]
```diff
@@ -476,6 +476,18 @@ func isIncorrectName(name string) bool {
...
 }
 
+// checkContainDotColumn checks field contains the table name.
+// for example :create table t (c1.c2 int default null).
+func (p *preprocessor) checkContainDotColumn(stmt *ast.CreateTableStmt) {
+	tName := stmt.Table.Name.String()
+	for _, colDef := range stmt.Cols {
+		// check table name.
+		if colDef.Name.Table.O != tName && len(colDef.Name.Table.O) != 0 {
+			p.err = ddl.ErrWrongTableName.GenByArgs(colDef.Name.Table.O)
```
Comment (zimulala) [inline plan/preprocess.go]: I think we can break here.

**Comment by ciscoxll** [comment]
@winkyao @zimulala  PTAL

**Pull request merged**